### DISCUSSION TITLE
How can we overwrite built-in components with custom components?

### DISCUSSION THREAD

**Discussion by yannxaver** [fuma-nama/fumadocs#1096] [Q&A]
https://nextjs.org/docs/pages/building-your-application/configuring/mdx#global-styles-and-components

```ts
import type { MDXComponents } from 'mdx/types';

export function useMDXComponents(
  components: MDXComponents,
): MDXComponents {
  return {
    img: (props) => <p>hello world</p>,
    h1: (props) => <p>hello world</p>,
    ...components,
  };
}
```

This does not work when using fumadocs.

**Comment by fuma-nama** [comment]
Please refer to the official docs instead of Next.js docs.

You can simply change the components prop in MDX renderer: https://fumadocs.vercel.app/docs/mdx#usage
